Document support of OIDC Implicit flow in Kibana. (#45693)
elastic/elasticsearch: x-pack/docs/en/security/authentication/oidc-guide.asciidoc

The Elastic Stack supports single sign-on (SSO) using OpenID Connect via {kib} using
{es} as the backend service that holds most of the functionality. {kib} and {es}

> together represent an OpenID Connect Relying Party (RP) that supports the Authorization Code Flow and Implicit Flow as these are defined in the OpenID Connect specification.

Reviewer: ```suggestion
together represent an OpenID Connect Relying Party (RP) that supports the authorization code Flow and implicit flow as these are defined in the OpenID Connect specification.
```

nor the Elastic Stack implementation impose any constraints on this value.
- `Redirect URI`: This is the URI where the OP will redirect the user's browser after authentication. The
appropriate value for this will depend on your setup and whether or not {kib} sits behind a proxy or

> load balancer. It will typically be +$\{kibana-url}/api/security/v1/oidc+ or +$\{kibana-url}/api/security/v1/oidc/implicit+

Reviewer 2: ```suggestion
load balancer. It will typically be +$\{kibana-url}/api/security/v1/oidc+ (for the authorization code flow) or +$\{kibana-url}/api/security/v1/oidc/implicit+
```

op.token_endpoint::
    The URL for the Token Endpoint in the OpenID Connect Provider. This is the endpoint where

>     {es} will send a request to exchange the code for an ID Token. Setting this is optional when

Reviewer: ```suggestion
    {es} will send a request to exchange the code for an ID Token.  This setting this is optional when
```

>     using the Implicit Flow. The value for this setting should be provided by your OpenID Connect Provider.

Reviewer: ```suggestion
    you use the implicit flow. The value for this setting should be provided by your OpenID Connect Provider.
```

{kib} has in-built protection against _Cross Site Request Forgery_ attacks, which
is designed to prevent the {kib} server from processing requests that
originated from outside the {kib} application.

> In case you use the authorization code flow, {kib} needs to support unsolicited messages that originate from your

Reviewer: ```suggestion
If you use the authorization code flow, {kib} needs to support unsolicited messages that originate from your
```

> OP or a third party (see <<third-party-login>>). In order to do so, we need to explicitly _whitelist_ the

Reviewer: Who is "we" in this case, Kibana? Or should it be changed to "you must explicitly..."?

> In order to do so, we need to ...

Author: >  Or should it be changed to "you must explicitly..."?
Yep, thanks!

> OP or a third party (see <<third-party-login>>). In order to do so, we need to explicitly _whitelist_ the

Author: ```suggestion
OP or a third party (see <<third-party-login>>). In order to do so, you must explicitly _whitelist_ the
```

{kib} has in-built protection against _Cross Site Request Forgery_ attacks, which
is designed to prevent the {kib} server from processing requests that
originated from outside the {kib} application.

> In order to support OpenID Connect Authorization Code flow messages that originate from your

Author: __note:__ we use `GET` for implicit flow and hence Kibana XSRF protection code isn't executed, so we don't need to whitelist anything here.

> In order to support OpenID Connect Authorization Code flow messages that originate from your

Reviewer 2: ```suggestion
In case you use the authorization code flow, Kibana needs to support unsolicited messages that originate from your
```

> OP or a third party (see <<third-party-login>>, we need to explicitly _whitelist_ the

Reviewer 2: OP or a third party (see <<third-party-login>>, we need to explicitly _whitelist_ the
=>
OP or a third party (see <<third-party-login>>. In order to do so, we need to explicitly _whitelist_ the
